### ISSUE TITLE
Tests uses MapLoader gives an error `thread "cached4" java.lang.NullPointerException`

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#7098]
I am getting this error at my local when I run test that uses MapLoader. For example:
https://github.com/hazelcast/hazelcast/blob/befbce0b9362329375e586facccdf18e17b989c0/hazelcast/src/test/java/com/hazelcast/map/nearcache/NearCacheTest.java#L504

```
Exception in thread "cached4" java.lang.NullPointerException
    at com.hazelcast.spi.impl.AbstractCompletableFuture$ExecutionCallbackRunnable.run(AbstractCompletableFuture.java:342)
    at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:212)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
    at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
    at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:92)
```

**Comment by mdogan** [comment]
`null` logger is passed to the `AbstractCompletableFuture`.

```
private LoadFinishedFuture() {
            super((Executor) null, null);
}
```

**Comment by ahmetmircik** [comment]
@mdogan https://github.com/hazelcast/hazelcast/pull/7092 is waiting for that.

**Comment by tombujok** [comment]
will be fixed once #7115 is merged

**Comment by ahmetmircik** [comment]
closed via https://github.com/hazelcast/hazelcast/pull/7115
